### ISSUE TITLE
[feature] Dao independent transactions

### ISSUE THREAD

**Issue by otopba** [vitusortner/floor#340]
Sometimes you need to make a transaction that will work with several DAO or will have a lot of logic. I would like to have separate methods for starting and ending a transaction. Now, having only an annotation, this is not possible

Thank you!

**Comment by vitusortner** [comment]
What would you think of something similar to the following? We could add a member function on `FloorDatabse` that can be used for executing DAO functions inside a transaction.

```dart
database.runInTransaction(() {
  // DAO invocations
});
```

**Comment by otopba** [comment]
@vitusortner 
Yes, it can be done that way. But then how can we say that the transaction is successful or not? If there was no exception, is the transaction successful?

**Comment by mqus** [comment]
Usually, yes. I don't know why that shouldn't be the case.

**Comment by mqus** [comment]
I want to add, though, that if we want to add support for `database.runInTransaction` then we will have to give the inner function its own database (backed by the transaction). So the actual invocation would look like this:
```dart
database.runInTransaction((transactionDB) async {
  // DAO invocations
  transactionDB.someDao.findEntriesInTable("value");
});
```
or this (just using a function reference instead of a closure):
```dart
database.runInTransaction(withTransaction);

Future<void> withTransaction(MyDB transactionDB) async{
  // DAO invocations
  transactionDB.someDao.findEntriesInTable("value");
}
```
This would fit nicely into #360, too.

**Comment by NikoBoerger** [comment]
Are there any updates on that? :)

**Comment by vitusortner** [comment]
Unluckily not. In case you're interested, feel free to open a PR that implements the desired functionality.
